tests/backuptestTriggers.go

```diff
@@ -0,0 +1,628 @@
+	for _, provider := range providers {
+		cloudCredUID := uuid.New()
+		cloudCredUidList = append(cloudCredUidList, cloudCredUID)
+		backupLocationUID = uuid.New()
+		credName := fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+		err := CreateCloudCredential(provider, credName, cloudCredUID, BackupOrgID, ctx)
+		if err != nil {
+			return err, "", *eventData
+		}
+		log.InfoD("Created Cloud Credentials with name - %s", credName)
+		customBackupLocationName := fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+		err = CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), BackupOrgID, "", true)
+		if err != nil {
+			return err, "", *eventData
+		}
+		backupLocationMap[backupLocationUID] = customBackupLocationName
+		log.InfoD("Created Backup Location with name - %s", customBackupLocationName)
+
+		eventData.BackupLocationName = customBackupLocationName
+		eventData.BackupLocationUID = backupLocationUID
+	}
+
+	return nil, "", *eventData
+}
+
+// Event for Source and Dest Cluster add
+func eventAddSourceAndDestinationCluster(inputsForEventBuilder *PxBackupLongevity) (error, string, EventData) {
+	defer GinkgoRecover()
+	eventData := &EventData{}
+
+	log.Infof("Adding Clusters for backup")
+	err := CreateApplicationClusters(BackupOrgID, "", "", ctx)
+	if err != nil {
+		return err, "", *eventData
+	}
+	clusterStatus, err := Inst().Backup.GetClusterStatus(BackupOrgID, SourceClusterName, ctx)
+	if err != nil {
+		return err, "", *eventData
+	}
+	if clusterStatus != api.ClusterInfo_StatusInfo_Online {
+		return fmt.Errorf("Cluster %s is not online", SourceClusterName), fmt.Sprintf("Cluster added but not online"), *eventData
+	}
+
+	clusterUid, err := Inst().Backup.GetClusterUID(ctx, BackupOrgID, SourceClusterName)
+	if err != nil {
+		return err, "", *eventData
+	}
+	eventData.ClusterUid = clusterUid
+
+	return nil, "", *eventData
+}
+
+// Event for Backup Creation
+func eventCreateBackup(inputsForEventBuilder *PxBackupLongevity) (error, string, EventData) {
+	defer GinkgoRecover()
+
+	eventData := &EventData{}
+	var backupNames []string
+
+	log.Infof("Creating a manual backup")
+	for _, namespace := range inputsForEventBuilder.BackupData.Namespaces {
+		backupName := fmt.Sprintf("%s-%v-%s", BackupNamePrefix, time.Now().Unix(), RandomString(10))
+		labelSelectors := make(map[string]string)
+		appContextsToBackup := FilterAppContextsByNamespace(inputsForEventBuilder.ApplicationData.SchedulerContext, []string{namespace})
+		err := CreateBackupWithValidation(
+			ctx,
+			backupName,
+			SourceClusterName,
+			inputsForEventBuilder.BackupData.BackupLocationName,
+			inputsForEventBuilder.BackupData.BackupLocationUID,
+			appContextsToBackup,
+			labelSelectors,
+			BackupOrgID,
+			inputsForEventBuilder.BackupData.ClusterUid, "", "", "", "")
+		if err != nil {
+			return err, "Error occurred while taking backup", *eventData
```

Comment:
you can format the error message with err and "Error occurred while taking backup" and remove string return type.
ex: fmt.Errorf("Error occurred while taking backup, error: %v",err)

Reply:
The empty string is used to get any abnormalities thus populated the same with the message in case of error.